packages/actor-rdf-join-wrap-stream/lib/ActorRdfJoinWrapStream.ts

```diff
@@ -0,0 +1,87 @@
+import type { MediatorIteratorTransform } from '@comunica/bus-iterator-transform';
+import type {
+  IActionRdfJoin,
+  IActorRdfJoinOutputInner,
+  IActorRdfJoinArgs,
+  MediatorRdfJoin }
+  from '@comunica/bus-rdf-join';
+import { ActorRdfJoin, KEY_CONTEXT_WRAPPED_RDF_JOIN } from '@comunica/bus-rdf-join';
+import type { IMediatorTypeJoinCoefficients } from '@comunica/mediatortype-join-coefficients';
+import type { IQueryOperationResultBindings, MetadataBindings } from '@comunica/types';
+import type * as RDF from '@rdfjs/types';
+import type { AsyncIterator } from 'asynciterator';
+
+/**
+ * A comunica Wrap Stream RDF Join Actor.
+ */
+export class ActorRdfJoinWrapStream extends ActorRdfJoin {
+  public readonly mediatorJoin: MediatorRdfJoin;
+  public readonly mediatorIteratorTransform: MediatorIteratorTransform;
+
+  public constructor(args: IActorRdfJoinWrapStreamArgs) {
+    super(args, {
+      logicalType: 'inner',
+      physicalName: 'wrap-stream',
+      limitEntries: 0,
+      limitEntriesMin: true,
+      canHandleUndefs: true,
+      isLeaf: false,
+    });
+  }
+
+  public override async test(action: IActionRdfJoin): Promise<IMediatorTypeJoinCoefficients> {
+    if (action.context.get(KEY_CONTEXT_WRAPPED_RDF_JOIN) === this) {
```

Comment:
I would advise against storing the actor inside the context.
Instead, you can store the operation that is being wrapped, and test based on that.